File: sandbox.client/front/ts/model/ClientDetailsToSave.ts
import {Phone} from "./Phone";
import {Gender} from "./Gender";
import {AddressInfo} from "./AddressInfo";

export class ClientDetailsToSave {
  public id: number | null/*long*/;
  public surname: string;
  public name: string;
  public patronymic: string;
  public gender: Gender;
  public birthdate: string;
  public charmId: number/*int*/;
  public registrationAddressInfo: AddressInfo;
  public factualAddressInfo: AddressInfo;
  public phones: Phone[];
  public deletedPhones: Phone[];
}

File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/ClientDetailsToSave.java
package kz.greetgo.sandbox.controller.model;

import java.util.ArrayList;
import java.util.List;
import kz.greetgo.sandbox.controller.model.Gender;
import kz.greetgo.sandbox.controller.model.Phone;
import kz.greetgo.sandbox.controller.model.RegistrationAddressInfo;
import kz.greetgo.sandbox.controller.model.ResidentialAddressInfo;

public class ClientDetailsToSave {
  public Long id;
  public String surname;
  public String lastname;
  public String patronymic;
  public Gender gender;
  public String birthdate;
  public int charmId;
  public RegistrationAddressInfo registrationAddressInfo;
  public ResidentialAddressInfo residentialAddressInfo;
  public List<Phone> phones = new ArrayList<>();
}
